src/components/icons/MoonIcon.astro

```diff
@@ -0,0 +1,48 @@
+---
+import type { Pixel } from '../../model/model';
+
+export interface Props {
+  readonly height: Pixel;
+  readonly width: Pixel;
+}
+
+const { height = '24px', width = '24px' } = Astro.props;
+---
+
+<svg
+  class="icon moon"
+  xmlns="http://www.w3.org/2000/svg"
+  height={height}
+  width={width}
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round">
+    <path
+      d="M21 12.79A9 9 0 0111.21 3a7 7 0 000 14 9 9 0 009.79-4.21z"
+    ></path>
+</svg>
+
+<style>
```

Comment:
for best practices, this icon should be standalone without style. to stylize this should be done from where it is used.